packages/salesforcedx-vscode-apex-debugger/test/unit/core/sessionService.test.ts

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfdxCommandBuilder } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { SessionService } from '../../../src/core/sessionService';
+import { CommandOutput } from '../../../src/utils/commandOutput';
+
+describe('Debugger session service', () => {
+  let service: SessionService;
+  const cmdWithArgSpy = sinon.spy(SfdxCommandBuilder.prototype, 'withArg');
+  const cmdWithFlagSpy = sinon.spy(SfdxCommandBuilder.prototype, 'withFlag');
+  const cmdBuildSpy = sinon.spy(SfdxCommandBuilder.prototype, 'build');
+
+  beforeEach(() => {
+    service = new SessionService();
+  });
+
+  describe('Helpers', () => {
+    it('Should detect an Apex Debugger session ID', () => {
+      expect(service.isApexDebuggerSessionId('07aFAKE')).to.equal(true);
+    });
+
+    it('Should not detect an Apex Debugger session ID', () => {
+      expect(service.isApexDebuggerSessionId('FAKE')).to.equal(false);
+    });
+  });
+
+  describe('Start', () => {
+    let origSpawn: any, mySpawn: any;
+    const childProcess = require('child_process');
+    const mockSpawn = require('mock-spawn');
+
+    beforeEach(() => {
+      origSpawn = childProcess.spawn;
+      mySpawn = mockSpawn();
+      childProcess.spawn = mySpawn;
+    });
+
+    afterEach(() => {
+      childProcess.spawn = origSpawn;
+      cmdWithArgSpy.reset();
+      cmdWithFlagSpy.reset();
+      cmdBuildSpy.reset();
+    });
+
+    it('Should start successfully', async () => {
+      mySpawn.setDefault(mySpawn.simple(0, '{"result":{"id":"07aFAKE"}}'));
+
+      const cmdOutput: CommandOutput = await service.start();
+
+      expect(cmdOutput.getStdOut()).to.equal('{"result":{"id":"07aFAKE"}}');
+      expect(cmdOutput.getId()).to.equal('07aFAKE');
+      expect(service.isConnected()).to.equal(true);
+      expect(service.getSessionId()).to.equal('07aFAKE');
+    });
+
+    it('Should build command', async () => {
+      mySpawn.setDefault(mySpawn.simple(0, '{"result":{"id":"07aFAKE"}}'));
+
+      await service
+        .forProject('project')
+        .withUserFilter('user')
+        .withRequestFilter('request')
+        .withEntryFilter('entry')
+        .start();
+
+      expect(cmdWithArgSpy.getCall(0).args).to.have.same.members([
```

Comment:
I'm worried about the maintainability of these types of assertions?

I couldn't find the equivalent of Hamcrest matchers for TypeScript/JavaScript but it seems like it would be better to have a different object do the comparisons instead of doing it with individual `expect` statements.

So, I don't have an actionable item at this point, but we should think about how we could do this more cleanly since we probably want to check the same thing for all the command calls.